### ISSUE TITLE
DOC: add additional gallery examples

### ISSUE THREAD

**Issue by jorisvandenbossche** [geopandas/geopandas#529]
Some ideas:

- make interactive plot with folium or ipyleaflet
- combine with rasterio (eg extracting values from raster based on Point series)
- examples with geoplot
- sjoin example
- ...

**Comment by Abdullium** [comment]
> Some ideas:
> 
> * [x]  example converting existing dataframe with x/y columns
> * [ ]  make interactive plot with folium or ipyleaflet
> * [ ]  combined plot of raster and vector data
> * [ ]  combine with rasterio (eg extracting values from raster based on Point series)
> * [x]  examples with geoplot
> * [ ]  sjoin example
> * [ ]  example using the spatial index (`.sindex`) explicitly

hmm. you want an example in the documentation about these are:- 
* adding a page in docs to construct a geodataframe from dataframe 
* adding gallery example for plotting data with folium

**Comment by Abdullium** [comment]
> > Some ideas:
> > 
> > * [x]   example converting existing dataframe with x/y columns
> > * [ ]   make interactive plot with folium or ipyleaflet
> > * [ ]   combined plot of raster and vector data
> > * [ ]   combine with rasterio (eg extracting values from raster based on Point series)
> > * [x]   examples with geoplot
> > * [ ]   sjoin example
> > * [ ]   example using the spatial index (`.sindex`) explicitly
> 
> hmm. you want an example in the documentation about these are:-
> 
> * adding a page in docs to construct a geodataframe from dataframe
> * adding gallery example for plotting data with folium

you can assign the task to me I write brilliant documentation.

**Comment by Moddy2024** [comment]
@jorisvandenbossche  I can help make interactive plot with ipyleaflet. Can you show some example of it ?

**Comment by martinfleis** [comment]
@Moddy2024 you can check how current examples stored in [`doc/source/gallery`](https://github.com/geopandas/geopandas/tree/main/doc/source/gallery) look like and how ipyleaflet examples on their documentation are designed.

**Comment by akshatvishu** [comment]
@martinfleis  Is showcasing an example with ` ipyleaflet` still open ?

**Comment by martinfleis** [comment]
@akshatvishu yes, we still don't have such an example in our data.

**Comment by akshatvishu** [comment]
> @akshatvishu yes, we still don't have such an example in our data.

Also, as `ipyleaflet ` is not installed as a part of `geopandas dev env` do, I need to `pip install` on top my example notebook ?

**Comment by martinfleis** [comment]
No, if the required package is not part of the environment, add it to the environment.

**Comment by akshatvishu** [comment]
> No, if the required package is not part of the environment, add it to the environment.

Ok! Also, would you prefer that the `dataset`  used in the example is from `geodatasets` or can I use any open-source dataset?

**Comment by martinfleis** [comment]
You can use external open datasets (if the license permits) if it can be fetched online as we can expect that the URL will not change. And it if it is not a large file to load. If you can stick to `geodatasets` that may be slightly preferable but not necessary.

**Comment by CreaTechJulius** [comment]
@martinfleis @jorisvandenbossche , im interested in contributing. I could do a gallery example for the combination with rasterio
("combine with rasterio (eg extracting values from raster based on Point series")

**Comment by ArbyC** [comment]
@martinfleis @jorisvandenbossche I'm interested in contributing to the gallery of example as well. Can I go about with combining raster and vector data plot and rasterio operations?

**Comment by notix3333** [comment]
Hi @martinfleis / @jorisvandenbossche,

I’d like to take the "ipyleaflet" gallery example from this issue.

My plan is to add a concise example showing how to display a GeoDataFrame interactively with `ipyleaflet`, using a dataset from `geodatasets` if possible. I’ll follow the existing examples in doc/source/gallery and add "pyleaflet" to the documentation environment if needed.

Please let me know if this direction still makes sense, and I can open a PR  :)

**Comment by martinfleis** [comment]
@notix3333 go ahead. Thanks!

**Comment by DavidVal2** [comment]
Hi! I'm a beginner coder looking for a first issue and I would like to work on this one. I plan on starting off simple, so I aim to create a new gallery example for sjoin

Edit: Hi! Quick update on my plan for this issue.

After checking the current `main` branch, I noticed that gallery examples for `sjoin`, `folium`, and `rasterio` already exist.  
So instead of adding a duplicate notebook, I’ll focus on improving the existing `spatial_joins` gallery example to make it clearer for beginners (intro/objectives/wording around join types and predicates).

If this looks good I can open a PR shortly!

**Comment by salvatoremellino81** [comment]
Hi all,

I noticed that ipyleaflet and spatial joins are already being addressed in recent comments.

I would be interested in contributing a small beginner-friendly improvement related to creating a GeoDataFrame from tabular x/y coordinates, with a clearer explanation of CRS assignment and a simple plotting/export workflow.

Would this still be useful for the gallery/documentation, or is there another documentation gap where a first contribution would be more helpful?

I have a GIS and geospatial data interpretation background, so I would be happy to start with a focused documentation contribution.

**Comment by Manacydev** [comment]
Hi, I'd like to work on adding a gallery example for sjoin (spatial join). Is this still open for contribution?

**Comment by pratikverse** [comment]
Hi, I'd really like to work on this. Is it still open for contribution?

**Comment by Catalyst097** [comment]
Hi! I'm a first-time contributor and would love to work on adding a gallery example for interactive plotting with Folium/ipyleaflet. Is this item available for me to take on?"

**Comment by Pixel20coder** [comment]
I'd like to add the combined plot of raster and vector data example, if that's still open — let me know if someone's already working on it.
